front/pages/api/v1/w/[wId]/data_sources/[name]/tables/[tId]/rows/index.ts

```diff
@@ -171,21 +171,19 @@ async function handler(
       const { truncate } = bodyValidation.right;
       let { rows: rowsToUpsert } = bodyValidation.right;
 
-      // Make every key in the rows lowercase and ensure there are no duplicates.
+      // Make sure every key in the rows are lowercase
       const allKeys = rowsToUpsert.map((row) => Object.keys(row.value)).flat();
-      const keysSet = new Set<string>();
-      for (const key of allKeys) {
-        if (keysSet.has(key)) {
-          return apiError(req, res, {
-            status_code: 400,
-            api_error: {
-              type: "invalid_request_error",
-              message: `Duplicate key: ${key}`,
-            },
-          });
-        }
-        keysSet.add(key);
+      if (!allKeys.every((key) => /^[a-z0-9_]+$/.test(key))) {
```

Comment:
`nit` you could use `isSlugified` helper.

Reply:
Thanks updating to this for consistency across code paths 👍